lib/Site.js

```diff
@@ -188,10 +188,14 @@ Site.prototype.createPageData = function (config) {
     baseUrlMap: this.baseUrlMap,
     content: '',
     pageTemplate: config.pageTemplate,
+    useFavIcon: config.useFavIcon,
     rootPath: this.rootPath,
     src: config.pageSrc,
     title: config.title || '',
     titlePrefix: config.titlePrefix,
+    favIconPath: (config.favIconPath !== undefined)
+      ? (path.relative(path.dirname(resultPath), path.join(this.outputPath, config.favIconPath)))
+      : undefined,
```

Comment:
- Favicon path is usually [root-relative](https://stackoverflow.com/questions/5559578/having-links-relative-to-root). Use `config.baseUrl`.
- Let's calculate this before `return new Page`.
  - Use `fs.existsSync` to check if the file exists, otherwise log a warning and set it to `undefined`.
  - If unspecified, default to `'favicon.ico'` if the file exists.